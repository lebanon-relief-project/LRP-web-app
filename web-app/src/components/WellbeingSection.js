import React from "react";
import styled from "styled-components";
import devices from "../styles/Devices";

import Card from "../components/Card";
import { Routes } from "../App.js";
import { StyledSection } from "../styles/GlobalStyles";
import People from "../assets/images/People.svg";

const WellbeingSection = () => {
  return (
    <StyledSection>
      <FlexContainer>
        <StyledPeople src={People} />

        <IllustrationAndCard>
          <ChallengeCard>
            <Card
              secondary="true"
              disabled
              title="Try our 14 days of wellbeing"
              path={Routes.HOME}
              buttonText="Coming soon"
            >
              <Paragraph>
                Keep your mind and body engaged by joining our 14 days of
                wellbeing. Created by our team of mental health experts, these
                exercises offer simple daily tasks which will help you manage
                negative emotions and reach a more positive and calmer mindset.
              </Paragraph>
            </Card>
          </ChallengeCard>
        </IllustrationAndCard>
      </FlexContainer>
    </StyledSection>
  );
};

const FlexContainer = styled.div`
  display: flex;
  flex-direction: row;

  @media (max-width: ${devices.mobile}) {
    flex-direction: column;
  }
`;

const StyledPeople = styled.img`
  max-width: 100%;
  object-fit: contain;
  width: auto;
  margin: 2%;

  @media (max-width: ${devices.ipad}) {
    width: 80%;
  }
  @media (max-width: ${devices.mobile}) {
    width: 100%;
    max-width: 100%;
    padding-left: 0rem;
  }
`;

const Paragraph = styled.p`
  font-family: Raleway;
  font-style: normal;

  font-size: 16px;
  line-height: 150%;
  color: #262626;
`;

const ChallengeCard = styled.div`
  display: flex;
  width: 85%;
  @media (max-width: ${devices.ipadpro}) {
    width: 50%;
    right: 0;
  }
  @media (max-width: ${devices.mobile}) {
    position: static;
    width: 100%;
  }
`;

const IllustrationAndCard = styled.div`
  @media (min-width: ${devices.mobile}) {
    display: flex;
    position: relative;
  }
`;

export default WellbeingSection;
