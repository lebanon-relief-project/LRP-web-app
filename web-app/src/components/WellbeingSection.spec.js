import React from "react";
import { render } from "@testing-library/react";
import WellbeingSection from "./WhatHappenedSection";
import "jest-styled-components";

describe("the WhatHappened component", () => {
  let container;
  it("should match the snapshot", () => {
    ({ container } = renderWhatHappenedSection());

    expect(container).toMatchSnapshot();
  });
});

const renderWhatHappenedSection = () => {
  return render(<WellbeingSection />);
};