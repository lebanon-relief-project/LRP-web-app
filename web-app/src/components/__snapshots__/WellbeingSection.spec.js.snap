// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the WellbeingSection component should match the snapshot 1`] = `
.c8 {
  background-color: #f0f0f0;
  color: black;
  font-style: normal;
  font-weight: bold;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-width: 25%;
  max-width: 25%;
  margin-bottom: 5%;
  padding: 0.5rem;
}

.c6 {
  font-family: Playfair Display;
  font-weight: 900;
  font-size: 38px;
  min-width: auto;
  line-height: 100%;
  padding: 0;
  text-align: left;
  margin-bottom: 0;
}

.c5 {
  position: relative;
  border: none;
  width: 100%;
  padding: 1rem 0;
  margin: 3rem 0rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Raleway;
  font-size: 16px;
  font-style: normal;
  font-weight: 400;
  line-height: 24px;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  text-align: left;
  background-color: white;
}

.c0 {
  background: #ffffff;
  margin: 0;
  margin-bottom: 5%;
  padding: 0 3rem;
  max-width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 {
  max-width: 100%;
  object-fit: contain;
  width: auto;
  margin: 2%;
}

.c7 {
  font-family: Raleway;
  font-style: normal;
  font-size: 16px;
  line-height: 150%;
  color: #262626;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 85%;
}

@media (max-width:576px) {
  .c8 {
    width: 50%;
    max-width: 100%;
  }
}

@media (max-width:576px) {
  .c6 {
    font-size: 30px;
    line-height: 38px;
  }
}

@media (max-width:1025px) {
  .c5 {
    width: 75%;
    width: 100%;
    min-width: 0;
  }
}

@media (max-width:768px) {
  .c5 {
    margin: 3rem 0;
  }
}

@media (max-width:576px) {
  .c5 {
    width: 90%;
    margin: 1rem auto;
  }
}

@media (max-width:576px) {
  .c0 {
    padding: 0;
    margin-top: 0;
  }
}

@media (max-width:576px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:768px) {
  .c2 {
    width: 80%;
  }
}

@media (max-width:576px) {
  .c2 {
    width: 100%;
    max-width: 100%;
    padding-left: 0rem;
  }
}

@media (max-width:1025px) {
  .c4 {
    width: 50%;
    right: 0;
  }
}

@media (max-width:576px) {
  .c4 {
    position: static;
    width: 100%;
  }
}

@media (min-width:576px) {
  .c3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    position: relative;
  }
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        class="c2"
        src="People.svg"
      />
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <fieldset
            class="c5"
            disabled=""
          >
            <legend
              class="c6"
              disabled=""
            >
              Try our 14 days of wellbeing
            </legend>
            <p
              class="c7"
            >
              Keep your mind and body engaged by joining our 14 days of wellbeing. Created by our team of mental health experts, these exercises offer simple daily tasks which will help you manage negative emotions and reach a more positive and calmer mindset.
            </p>
            <a
              class="c8"
              disabled=""
              href="/"
              secondary="true"
            >
              Coming soon
            </a>
          </fieldset>
        </div>
      </div>
    </div>
  </section>
</div>
`;
